### PR TITLE
Calvin/add signin errors

### PR DESCRIPTION
**Pull request by calvind777**
adds some error text in the bottom of the modal if your sign in doesn't go through

Eric todos:
- [x] update the error text
- [x] allow submit on enter key

### CODE REVIEW

File: app/assets/javascripts/components/home/home_login.jsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ class HomeLogin extends DefaultForm {
 
   constructor(props) {
     super(props);
-    this.state = { showModal: false };
+    this.state = { showModal: false, hideErrors: true };
```
Comment (eric-ls) [inline app/assets/javascripts/components/home/home_login.jsx]: nit: put on separate lines

File: app/assets/javascripts/components/home/home_login.jsx [hunk 2]
```diff
@@ -41,7 +51,7 @@ class HomeLogin extends DefaultForm {
             <Modal.Title>Log In</Modal.Title>
           </Modal.Header>
           <form
```
Comment (eric-ls) [inline app/assets/javascripts/components/home/home_login.jsx]: can you remove the `<form>` tag and put all the logic in the `onClick` handler?

File: app/assets/javascripts/components/home/home_login.jsx [hunk 3]
```diff
@@ -23,6 +23,16 @@ class HomeLogin extends DefaultForm {
...
   }
 
+  _handleLogin = () => {
+    const success = () => {
+      window.location = "/dashboard";
+    }
+    const failure = () => {
+      this.setState({hideErrors: false});
+    }
+    Requester.post(APIConstants.sessions.create, {'email':this.state.email,'password':this.state.password}, success, failure);
```
Comment (eric-ls) [inline app/assets/javascripts/components/home/home_login.jsx]: you can probably use `this._formFields()` since this class inherits from `DefaultForm`? idk

**Comment by calvind777** [comment]
@eric-ls fixed!

**Comment by eric-ls** [comment]
yay almost there :)

one more issue @calvind777: when I try to login with an email that doesn't exist (or an email that's incorrectly formatted), there's a 500 error:
screen shot 2017-01-19 at 9 50 39 pm

Do you think you can check this out?

**Comment by calvind777** [comment]
@eric-ls fixed i think

**Comment by eric-ls** [comment]
looks good @calvind777! 👍

**Pull request merged**